src/haxe/crypto/Sha512.hx

```diff
@@ -12,8 +12,8 @@ class Sha512
     public function new() {
 	}
 
-	public static function encode( s:String, ?encoding : haxe.io.Encoding ) : String {
-		var data = haxe.io.Bytes.ofString(s, encoding);
+	public static function encode( s:String #if !haxe3 , ?encoding : haxe.io.Encoding #end ) : String {
+		var data = haxe.io.Bytes.ofString(s #if !haxe3 , encoding #end );
```

Comment:
```suggestion
		var data = haxe.io.Bytes.ofString(s #if haxe4 , encoding #end );
```

Reply:
(Sorry seems like I should have made a batch, first time I use this resolve feature)